### PR TITLE
Let users define segments using SQL

### PR DESCRIPTION
**Pull request by SeyZ**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [x] Create automatic tests
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

File: services/resources-getter.js [hunk 1]
```diff
@@ -63,66 +65,88 @@ function ResourcesGetter(model, opts, params) {
...
+            model.scope(segmentScope).findAll(findAllOpts)
+          ]);
+        } else {
+          return P.all([
+            model.unscoped().count(countOpts),
+            model.unscoped().findAll(findAllOpts)
+          ]);
+        }
+    })
+    .catch(function (err) {
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: `err` => `error`

File: services/resources-getter.js [hunk 2]
```diff
@@ -63,66 +65,88 @@ function ResourcesGetter(model, opts, params) {
...
+        new LiveQueryChecker().perform(rawQuery);
+
+        opts.connections[0]
+          .query(rawQuery, { type: opts.sequelize.QueryTypes.SELECT })
+          .then(function (result) {
+            result = result.map(function (r) { return r.id; } );
+            where[OPERATORS.AND].push({ id: { in: result } } );
+
+            return resolve(where);
+          }, function (err) {
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: `err` => `error`

File: services/resources-getter.js [hunk 3]
```diff
@@ -63,66 +65,88 @@ function ResourcesGetter(model, opts, params) {
...
+      if (segmentWhere) {
+        where[OPERATORS.AND].push(segmentWhere);
+      }
 
-    return where;
+      if (params.query) {
+        var rawQuery = params.query.trim();
+        new LiveQueryChecker().perform(rawQuery);
+
+        opts.connections[0]
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: What if the connection for the collection is the first one?

File: services/resources-getter.js [hunk 4]
```diff
@@ -63,66 +65,88 @@ function ResourcesGetter(model, opts, params) {
...
+          .query(rawQuery, { type: opts.sequelize.QueryTypes.SELECT })
+          .then(function (result) {
+            result = result.map(function (r) { return r.id; } );
+            where[OPERATORS.AND].push({ id: { in: result } } );
```
Comment (arnaudbesnier) [inline services/resources-getter.js]: Same comment as for the Rails liana.
Is it really scalable? What if my SQL query returns X millions of ids?

File: services/resources-getter.js (hunk 3 above)
Comment (SeyZ) [reply services/resources-getter.js]: I've added the same comment than in the query-stat-getter

**Pull request merged**